spec/emerald/parser_spec.rb

```diff
@@ -100,4 +100,58 @@
                           Emerald::Number.new(30)])
     end
   end
+
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"' )])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+
+    it "parses a string with a set of quotation marks are immediately after a backslash" do
+      source = '"Say \"hello\""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Say \"hello\""')])
+    end
+
+    it "parses a string over two lines" do
+      source = '"This is a multiline \n  string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is a multiline \n  string"')])
+    end
+
+    it "parses a multiline string" do
+      source = '"This is another
+      string across
+      three lines"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is another
+      string across
+      three lines"')])
+    end
+  end
```

Comment:
Ah, I was wrong. It matches strings that are the last thing on the line. Try parsing two strings on one line. Probably always good to have a test where we parse more than one.